### ISSUE TITLE
Unhandled top-level exception: Merkle_mask.Masking_merkle_tree.Make(Inputs).Attached.Dangling_parent_reference

### ISSUE THREAD

**Issue by gregbostrom** [MinaProtocol/mina#6949]
After installing version
```
Commit [DIRTY]9652f8ee092ea77e29f5ab49fa0a295e36743e8b on branch HEAD
```
Had this crash after initial coda daemon start.

coda daemon output:
```
2020-12-04 20:40:32 UTC [Fatal] Unhandled top-level exception: $exn
Generating crash report
	exn: {
  "sexp": [
    "Merkle_mask.Masking_merkle_tree.Make(Inputs).Attached.Dangling_parent_reference(\"2f54127f-9dc1-3256-a755-8d31ad22fbd9\")"
  ],
  "backtrace": [
    "Raised at file \"src/lib/merkle_mask/masking_merkle_tree.ml\", line 101, characters 10-50",
    "Called from file \"src/lib/merkle_mask/masking_merkle_tree.ml\", line 513, characters 6-26",
    "Called from file \"src/lib/network_pool/transaction_pool.ml\", line 903, characters 24-75",
    "Called from file \"src/lib/network_pool/transaction_pool.ml\", line 906, characters 26-68",
    "Called from file \"src/lib/network_pool/transaction_pool.ml\", line 1143, characters 18-30",
    "Called from file \"src/lib/network_pool/network_pool_base.ml\", line 93, characters 15-67",
    "Called from file \"src/lib/pipe_lib/strict_pipe.ml\", line 152, characters 28-37",
    "Called from file \"src/deferred0.ml\", line 56, characters 64-69",
    "Called from file \"src/job_queue.ml\" (inlined), line 131, characters 2-5",
    "Called from file \"src/job_queue.ml\", line 171, characters 6-47"
  ]
}


  ☠  Coda Daemon crashed.
   The Coda Protocol developers would like to know why!
```

**Comment by gregbostrom** [comment]
Updated to version:
```
Commit [DIRTY]880882e45e48aaf987aca191f488c7f65609c305 on branch HEAD
```

Similiar result:
```
2020-12-06 22:40:27 UTC [Info] Peer $peer didn't have enough information to answer ledger_hash query. See error for more details: $error
	peer: {
  "host": "49.50.96.77",
  "peer_id": "12D3KooWPpLHkA8uE7wvkbKR9A1ZvmQexG5bQAxsE6rp9ZtXtbXV",
  "libp2p_port": 8302
}
	error: {
  "sexp":
    "Refused to answer_query for ledger_hash: 27933808421117605816496510522425823357234922528944726275254018875284526352842"
}
2020-12-06 22:40:27 UTC [Info] None of the peers contacted were able to answer ledger_hash query -- trying more
2020-12-06 22:40:27 UTC [Fatal] Unhandled top-level exception: $exn
Generating crash report
	exn: {
  "sexp": [
    "Merkle_mask.Masking_merkle_tree.Make(Inputs).Attached.Dangling_parent_reference(\"e9058dab-9fb3-355b-13df-119100ed75d6\", \"File \\\"src/lib/transition_router/transition_router.ml\\\", line 505, characters 61-68\")"
  ],
  "backtrace": [
    "Raised at file \"src/lib/merkle_mask/masking_merkle_tree.ml\", line 104, characters 10-57",
    "Called from file \"src/lib/merkle_mask/masking_merkle_tree.ml\", line 516, characters 6-26",
    "Called from file \"src/lib/network_pool/transaction_pool.ml\", line 903, characters 24-75",
    "Called from file \"src/lib/network_pool/transaction_pool.ml\", line 906, characters 26-68",
    "Called from file \"src/lib/network_pool/transaction_pool.ml\", line 1143, characters 18-30",
    "Called from file \"src/lib/network_pool/network_pool_base.ml\", line 101, characters 15-67",
    "Called from file \"src/lib/pipe_lib/strict_pipe.ml\", line 152, characters 28-37",
    "Called from file \"src/deferred0.ml\", line 56, characters 64-69",
    "Called from file \"src/job_queue.ml\" (inlined), line 131, characters 2-5",
    "Called from file \"src/job_queue.ml\", line 171, characters 6-47"
  ]
}

```

**Comment by mrmr1993** [comment]
This was fixed some time ago in #7376. Closing.
